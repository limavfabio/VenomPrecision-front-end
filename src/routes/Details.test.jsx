// import React from 'react';
// import { render } from '@testing-library/react';
// import { Provider } from 'react-redux';
// import { RouterProvider } from 'react-router-dom';
// import configureStore from 'redux-mock-store';
// import { ToastContainer } from 'react-toastify';
// import Details from './Details';
// import router from '../router';

// // Mock the Redux store
// const mockStore = configureStore([]);

// describe('Details component', () => {
//   it('renders product details correctly', () => {
//     // Mock the product data
//     const mockProduct = {
//       name: 'Test Product',
//       description: 'Test description',
//       price: 10,
//       created_at: Date.now(),
//       image: 'test-image-url',
//     };

//     // Create a mock Redux store with the product data
//     const store = mockStore({
//       products: {
//         value: mockProduct,
//       },
//     });

//     // Render the Details component within the mocked store and router
//     const { getByText, getByAltText } = render(
//       <Provider store={store}>
//         <ToastContainer />
//         <RouterProvider router={router} />
//       </Provider>
//     );

//     // <MemoryRouter initialEntries={['/products/1']}>
//     //   <Route path="/products/:productId">
//     //     <Details />
//     //   </Route>
//     // </MemoryRouter>


//     // Assert that the product details are rendered correctly
//     expect(getByText(mockProduct.name)).toBeInTheDocument();
//     expect(getByText(mockProduct.description)).toBeInTheDocument();
//     expect(getByText(`$${mockProduct.price}`)).toBeInTheDocument();
//     expect(getByAltText(mockProduct.name)).toHaveAttribute('src', mockProduct.image);
//   });
// });
