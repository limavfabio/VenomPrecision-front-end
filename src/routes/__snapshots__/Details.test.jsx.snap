// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Home expects to render a snapshot of the Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-between bg-lime-500 px-4 py-2 md:hidden"
      >
        <button
          class=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="h-8 text-white"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <p
          class="font-bold text-white"
        >
          VenomPrecision
        </p>
        <div />
      </div>
      <div
        class="flex"
      >
        <div
          class="hidden h-screen min-w-max flex-col border border-gray-100 bg-white md:flex"
        >
          <div
            class="flex mt-10 justify-center"
          >
            <img
              alt="logo"
              class="w-40"
              src="/venom-precision-logo.png"
            />
          </div>
          <ul
            class="mt-10"
          >
            <a
              href="/"
            >
              <li
                class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold bg-lime-500 text-white"
              >
                <span
                  class="mr-5 text-xl"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 2a1 1 0 000 2h2a1 1 0 100-2H9z"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M4 5a2 2 0 012-2 3 3 0 003 3h2a3 3 0 003-3 2 2 0 012 2v11a2 2 0 01-2 2H6a2 2 0 01-2-2V5zm3 4a1 1 0 000 2h.01a1 1 0 100-2H7zm3 0a1 1 0 000 2h3a1 1 0 100-2h-3zm-3 4a1 1 0 100 2h.01a1 1 0 100-2H7zm3 0a1 1 0 100 2h3a1 1 0 100-2h-3z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                MAIN PAGE
              </li>
            </a>
            <a
              href="/products"
            >
              <li
                class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold text-slate-700"
              >
                <span
                  class="mr-5 text-xl"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.5 3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5v-9a.5.5 0 0 1 .5-.5h13zm-13-1A1.5 1.5 0 0 0 0 3.5v9A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-9A1.5 1.5 0 0 0 14.5 2h-13z"
                    />
                    <path
                      d="M7 5.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 1 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0zM7 9.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 0 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0z"
                    />
                  </svg>
                </span>
                ALL PRODUCTS
              </li>
            </a>
            <a
              href="/login"
            >
              <li
                class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold text-slate-700"
              >
                <span
                  class="mr-5 text-xl"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m13 16 5-4-5-4v3H4v2h9z"
                    />
                    <path
                      d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
                    />
                  </svg>
                </span>
                LOGIN
              </li>
            </a>
          </ul>
        </div>
        <div
          class="relative flex mt-5 sm:mt-0 flex-col sm:flex-row w-full justify-between items-center"
        >
          <div
            class="mb-5 sm:mb-0 flex-1"
          >
            <img
              alt=""
              class="w-2/3 mx-auto"
              src=""
            />
          </div>
          <div
            class="flex h-full w-full sm:w-1/3 min-w-fit flex-col justify-center px-5 text-end"
          >
            <h2
              class="mb-1 text-4xl"
            >
               
               
            </h2>
            <p
              class="mb-5"
            >
               
               
            </p>
            <div
              class="mb-5 flex flex-col justify-between"
            >
              <div
                class="flex justify-between px-4 py-2 text-sm bg-gray-300"
              >
                <p>
                  Package price
                </p>
                <p />
              </div>
              <div
                class="flex justify-between px-4 py-2 text-sm "
              >
                <p>
                  Date
                </p>
                <p>
                  Invalid Date
                </p>
              </div>
            </div>
            <a
              href="/products"
            >
              <p
                class="flex items-center self-end font-bold"
              >
                DISCOVER MORE MODELS
                 
                <svg
                  aria-hidden="true"
                  class="h-5 text-[#97BF0F]"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.25 4.5l7.5 7.5-7.5 7.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                 
              </p>
            </a>
          </div>
          <button
            class="bg-[#97BF0F] pl-8 pr-2 py-4 rounded-r-full absolute left-0 bottom-0 m-5"
            type="button"
          >
            <svg
              class="text-white rotate-180 text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-between bg-lime-500 px-4 py-2 md:hidden"
    >
      <button
        class=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="h-8 text-white"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
      <p
        class="font-bold text-white"
      >
        VenomPrecision
      </p>
      <div />
    </div>
    <div
      class="flex"
    >
      <div
        class="hidden h-screen min-w-max flex-col border border-gray-100 bg-white md:flex"
      >
        <div
          class="flex mt-10 justify-center"
        >
          <img
            alt="logo"
            class="w-40"
            src="/venom-precision-logo.png"
          />
        </div>
        <ul
          class="mt-10"
        >
          <a
            href="/"
          >
            <li
              class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold bg-lime-500 text-white"
            >
              <span
                class="mr-5 text-xl"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 2a1 1 0 000 2h2a1 1 0 100-2H9z"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M4 5a2 2 0 012-2 3 3 0 003 3h2a3 3 0 003-3 2 2 0 012 2v11a2 2 0 01-2 2H6a2 2 0 01-2-2V5zm3 4a1 1 0 000 2h.01a1 1 0 100-2H7zm3 0a1 1 0 000 2h3a1 1 0 100-2h-3zm-3 4a1 1 0 100 2h.01a1 1 0 100-2H7zm3 0a1 1 0 100 2h3a1 1 0 100-2h-3z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              MAIN PAGE
            </li>
          </a>
          <a
            href="/products"
          >
            <li
              class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold text-slate-700"
            >
              <span
                class="mr-5 text-xl"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14.5 3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5v-9a.5.5 0 0 1 .5-.5h13zm-13-1A1.5 1.5 0 0 0 0 3.5v9A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-9A1.5 1.5 0 0 0 14.5 2h-13z"
                  />
                  <path
                    d="M7 5.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 1 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0zM7 9.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 0 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0z"
                  />
                </svg>
              </span>
              ALL PRODUCTS
            </li>
          </a>
          <a
            href="/login"
          >
            <li
              class="w-64 ml-2 py-4 pl-3 flex  items-center  font-bold text-slate-700"
            >
              <span
                class="mr-5 text-xl"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m13 16 5-4-5-4v3H4v2h9z"
                  />
                  <path
                    d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
                  />
                </svg>
              </span>
              LOGIN
            </li>
          </a>
        </ul>
      </div>
      <div
        class="relative flex mt-5 sm:mt-0 flex-col sm:flex-row w-full justify-between items-center"
      >
        <div
          class="mb-5 sm:mb-0 flex-1"
        >
          <img
            alt=""
            class="w-2/3 mx-auto"
            src=""
          />
        </div>
        <div
          class="flex h-full w-full sm:w-1/3 min-w-fit flex-col justify-center px-5 text-end"
        >
          <h2
            class="mb-1 text-4xl"
          >
             
             
          </h2>
          <p
            class="mb-5"
          >
             
             
          </p>
          <div
            class="mb-5 flex flex-col justify-between"
          >
            <div
              class="flex justify-between px-4 py-2 text-sm bg-gray-300"
            >
              <p>
                Package price
              </p>
              <p />
            </div>
            <div
              class="flex justify-between px-4 py-2 text-sm "
            >
              <p>
                Date
              </p>
              <p>
                Invalid Date
              </p>
            </div>
          </div>
          <a
            href="/products"
          >
            <p
              class="flex items-center self-end font-bold"
            >
              DISCOVER MORE MODELS
               
              <svg
                aria-hidden="true"
                class="h-5 text-[#97BF0F]"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.25 4.5l7.5 7.5-7.5 7.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
               
            </p>
          </a>
        </div>
        <button
          class="bg-[#97BF0F] pl-8 pr-2 py-4 rounded-r-full absolute left-0 bottom-0 m-5"
          type="button"
        >
          <svg
            class="text-white rotate-180 text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
